rllib/core/models/tf/heads.py

```diff
@@ -18,6 +20,81 @@
 tf1, tf, tfv = try_import_tf()
 
 
+def auto_fold_unfold_time(input_spec: str):
```

Comment:
This method solve the problem that a recurrent encoder will take in [B, T, ...] observations and produce [B, T, ...] encodings. Therefore, we need to reshape these before/after we use our heads.